observer/holder/nodesHolder.go

```diff
@@ -206,14 +203,33 @@ func getNodesListAsString(nodes []*data.NodeData) string {
 	return strings.TrimSuffix(addressesString, ", ")
 }
 
+func cloneNodesSlice(input []*data.NodeData) []*data.NodeData {
+	clonedSlice := make([]*data.NodeData, len(input))
+	for idx, node := range input {
+		clonedSlice[idx] = &data.NodeData{
```

Comment:
this works but it is not future changes proof
Could have been:
```GO
clonedNodeData := *node
clonedSlice[idx] = &clonedNodeData 
```

Reply:
right. done